### PR TITLE
[UI] Add secret features.

### PR DESCRIPTION
**Pull request by Antoni-Czaplicki**

### CODE REVIEW

File: app/src/main/res/layout/contributors_activity.xml [hunk 1]
```diff
@@ -39,15 +47,30 @@
...
+                            android:layout_height="239dp">
+
+                            <ImageView
+                                android:id="@+id/glove"
+                                android:layout_width="144dp"
+                                android:layout_height="144dp"
+                                android:layout_gravity="center"
+                                android:layout_marginBottom="5dp"
+                                android:visibility="gone"
+                                app:srcCompat="@drawable/thanos_idle" />
```
Comment (kuba2k2) [inline app/src/main/res/layout/contributors_activity.xml:61]: ~~but that is this? does it relate to the app somehow?~~

File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt [hunk 2]
```diff
@@ -35,6 +36,61 @@ class ContributorsActivity : AppCompatActivity(), CoroutineScope {
...
+                else
+                    konami = 0
+                true
+            }
+            else -> {
+                konami = 0
+                super.onKeyUp(keyCode, event)
+            }
+        }
+    }
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt:92]: This is extreeemally long.. isn't there any easier, or shorter way to write this algorithm?

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/contributors_activity.xml (hunk 1 above)
Comment (Antoni-Czaplicki) [reply app/src/main/res/layout/contributors_activity.xml:61]: Thanos' gauntlet as a symbol of huge power

File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt (hunk 2 above)
Comment (Antoni-Czaplicki) [reply app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt:92]: There is a shorter way

File: app/src/main/res/layout/contributors_activity.xml [hunk 3]
```diff
@@ -85,16 +109,18 @@
...
             <pl.szczodrzynski.edziennik.ui.modules.base.lazypager.LazyViewPager
                 android:id="@+id/viewPager"
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
-                android:visibility="gone"
                 app:layout_behavior="@string/appbar_scrolling_view_behavior"
-                tools:visibility="visible" />
+                tools:visibility="invisible" >
+
+            </pl.szczodrzynski.edziennik.ui.modules.base.lazypager.LazyViewPager>
```
[the remark this answers is not in the document]
Comment (kuba2k2) [reply app/src/main/res/layout/contributors_activity.xml]: Are you sure? As far as I can remember, there is no blank line before the closing tag. There also is an unnecessary space after "invisible" which just looks bad.

File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt (hunk 2 above)
Comment (kuba2k2) [reply app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt]: Cool, I'm waiting then.

File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt [hunk 4]
```diff
@@ -35,6 +35,20 @@ class ContributorsActivity : AppCompatActivity(), CoroutineScope {
...
+    override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
+        return when (keyCode) {
+            KeyEvent.KEYCODE_DPAD_UP -> {konami = if (konami == 0 || konami == 1) konami.inc() else 0; true}
+            KeyEvent.KEYCODE_DPAD_DOWN -> {konami = if (konami == 2 || konami == 3) konami.inc() else 0; true}
+            KeyEvent.KEYCODE_DPAD_LEFT -> {konami = if (konami == 4 || konami == 6) konami.inc() else 0; true}
+            KeyEvent.KEYCODE_DPAD_RIGHT -> {konami = if (konami == 5 || konami == 7) konami.inc() else 0; true}
+            KeyEvent.KEYCODE_B -> {konami = if (konami == 8) konami.inc() else 0; true}
+            KeyEvent.KEYCODE_A -> {if (konami == 9) {konami += 1; b.konami.isVisible = true} else {konami = 0}; true}
+            else -> {konami = 0; super.onKeyUp(keyCode, event)}
+        }
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/settings/contributors/ContributorsActivity.kt:49]: Right, it is indeed shorter, but needs to be formatted somehow. Curly braces glued to the letters just don't look nice. Also, you're using `.inc()` and then suddenly `+= 1`. You can rethink this piece a bit more clever, somehow.

File: app/src/main/res/layout/contributors_activity.xml [hunk 5]
```diff
@@ -69,6 +96,7 @@
                             android:gravity="center"
                             android:text="@string/contributors_headline"
                             android:textAppearance="@style/NavView.TextView.Medium" />
+
```
Comment (kuba2k2) [inline app/src/main/res/layout/contributors_activity.xml:99]: Same here, no blank lines before closing tags.

Comment (kuba2k2) [reply app/src/main/res/layout/contributors_activity.xml:99]: The best idea is to just remove every blank line in the entire file, then let AS format it for you.

File: app/src/main/res/layout/contributors_activity.xml [hunk 6]
```diff
@@ -39,15 +47,34 @@
...
                             android:layout_height="wrap_content"
-                            android:layout_gravity="center_horizontal"
-                            android:layout_marginTop="16dp"
-                            android:scaleType="center"
-                            android:scaleX="0.8"
-                            android:scaleY="0.8"
-                            android:src="@mipmap/ic_splash" />
+                            android:layout_gravity="center"
+                            android:layout_marginTop="25dp"
+                            android:layout_marginBottom="25dp">
```
Comment (kuba2k2) [inline app/src/main/res/layout/contributors_activity.xml:55]: The image had 16dp top margin previously, please leave it this way. If you need more spacing for `@+id/glove`, add it to that element instead - this way it won't affect normal user's experience.

File: app/src/main/res/layout/contributors_activity.xml [hunk 7]
```diff
@@ -20,6 +20,14 @@
             android:layout_height="match_parent"
             android:fitsSystemWindows="true">
 
+            <ImageView
+                android:id="@+id/konami"
+                android:layout_width="90dp"
+                android:layout_height="90dp"
+                android:layout_gravity="bottom|end"
+                android:visibility="visible"
```
Comment (kuba2k2) [inline app/src/main/res/layout/contributors_activity.xml:28]: Visible by default?

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

**Review by kuba2k2 — APPROVED** [review]

**Pull request merged**